### PR TITLE
test for just one function right now

### PR DESCRIPTION
**Pull request by briansongdev**

### CODE REVIEW

File: scripts/analysis_helpers.py [hunk 1]
```diff
@@ -2,6 +2,26 @@
 from collections import defaultdict
 from all_helpers import *
 
+def check_consistent(ui, vi, uj, vj, mapping):
+    if (((ui, vi) in mapping and (uj, vj) in mapping) or ((ui, vj) in mapping and (uj, vi) in mapping)):
```
Comment (wangpatrick57) [inline scripts/analysis_helpers.py:6]: you can just do return ((ui, vi) in ...)

File: scripts/analysis_helpers.py [hunk 2]
```diff
@@ -2,6 +2,26 @@
...
+# Check for consistency using mapList1 and mapList2.
+def num_consistent_edges(left_edges, right_edges, mapping):
+    consistent_edges = 0
+    counter = 0
+    for i in left_edges[0]:
+        for j in right_edges[0]:
+            ui = i[0]
+            uj = i[1]
+            vi = j[0]
+            vj = j[1]
```
Comment (wangpatrick57) [inline scripts/analysis_helpers.py:19]: nice variables here, but an even better way is to do "for ui, uj in left_edges[0]" and same for the right_edges line

File: scripts/node_pair_extraction_helpers.py [hunk 3]
```diff
@@ -3,6 +3,62 @@
 from file_helpers import *
 from ortholog_helpers import *
 
+def add_valid_edges(list_of_species_nodes, lines):
```
Comment (wangpatrick57) [inline scripts/node_pair_extraction_helpers.py:6]: valid is too vague, either rename or define "valid" on top of the function in a comment

File: scripts/node_pair_extraction_helpers.py [hunk 4]
```diff
@@ -3,6 +3,62 @@
...
+    num_edges = 0
+    edges = set()
+    for i in list_of_species_nodes:
+        if(i[0] in lines and i[1] in lines):
```
Comment (wangpatrick57) [inline scripts/node_pair_extraction_helpers.py:10]: what does the "lines" variable represent? why would nodes be in "lines"?

File: scripts/node_pair_extraction_helpers.py [hunk 5]
```diff
@@ -3,6 +3,62 @@
...
+    processed_list_of_nodes = list()
+    counter = 0
+    for i in preprocessed_nodes[0]:
+        if i != "0" and preprocessed_nodes[1][counter] != "0":
+            if not(left_node_frequencies[i] >= 1 or int(right_node_frequencies[preprocessed_nodes[1][counter]]) >= 1):
+                processed_list_of_nodes.append((i, preprocessed_nodes[1][counter]))
+        counter += 1
+    return processed_list_of_nodes
+
+def remove_dupes(list_of_nodes):
```
Comment (wangpatrick57) [inline scripts/node_pair_extraction_helpers.py:34]: if the above is "remove dupes from dictionary", you should name this "remove dupes from list"

File: scripts/node_pair_extraction_helpers.py [hunk 6]
```diff
@@ -3,6 +3,62 @@
...
+    temporary_list = list_of_nodes_from_species(name)
+
+    reformatted_nodes = []
+    for i in temporary_list:
+        reformatted_nodes.append(map(str, i))
+    result = list(map(list, zip(*reformatted_nodes)))
+
+    processed_list = remove_dupes(result)
+
+    return processed_list
```
Comment (wangpatrick57) [inline scripts/node_pair_extraction_helpers.py:60]: nice function here, very readable because of all the subfunctions

**Review by patrickhyw — CHANGES_REQUESTED** [review]

**Pull request closed without merging**